http-common/src/connector.rs

```diff
@@ -768,8 +767,37 @@ fn get_num_systemd_sockets() -> Result<Option<std::os::unix::io::RawFd>, String>
         }
     }
 
-    // Return the number of systemd sockets.
-    Ok(Some(listen_fds))
+    // If there is more than one socket, this is edged. We can attempt to match the socket name to systemd.
+    // This happens when a unix Uri is provided in the config.toml. Systemd sockets get created noneless so we still prefer to use them.
+    // If no socket name is provided, this mean edged is telling us there is not matching systemd socket, we return early. See above "((listen_fds > 1)&&(socket_name == None))"
+    // If a socket name is provided but we don't see the env variable LISTEN_FDNAMES, it means we are probably on an older OS, we can't match either.
+    let listen_fdnames = match get_env("LISTEN_FDNAMES")? {
+        Some(listen_fdnames) => listen_fdnames,
+        None => return Ok(None),
+    };
+    let listen_fdnames: Vec<&str> = listen_fdnames.split(':').collect();
+
+    if listen_fds != listen_fdnames.len() as i32 {
+        return Err(format!(
+            "Mismatch, there are {} fds, and {} names",
+            listen_fds,
+            listen_fdnames.len()
+        ));
+    }
+
+    let mut fd_index = SD_LISTEN_FDS_START;
```

Comment:
This can be just:

```rust
if let Some(index) = listen_fdnames.iter().position(|fdname| fdname == socket_name) {
    Ok(SD_LISTEN_FDS_START + index as _)
}
else {
    Err(format!("could not find a match for...")
}
```

(The `as _` might make clippy complain about `cast_sign_loss`. In that case just `#[allow]` it on this line. You can search for `cast_sign_loss` in this repo for examples.)

Reply:
It looks better, thanks a lot